state/block_test.go

```diff
@@ -40,29 +38,111 @@ func TestGetBlockByNumber(t *testing.T) {
 
 	// Set the block's header & blockData in the blockState
 	// SetHeader also sets mapping [blockNumber : hash] in DB
-	err = stateService.Block.SetHeader(*blockHeader)
-	if err != nil {
-		t.Fatal(err)
-	}
+	err = stateService.Block.SetHeader(blockHeader)
+	require.Nil(t, err)
 
 	err = stateService.Block.SetBlockData(hash, blockData)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.Nil(t, err)
 
 	// Get block & check if it's the same as the expectedBlock
 	expectedBlock := types.Block{
 		Header: blockHeader,
 		Body:   blockBody,
 	}
 	retBlock, err := stateService.Block.GetBlockByNumber(blockHeader.Number)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.Nil(t, err)
+
 	retBlock.Header.Hash()
 
-	if !reflect.DeepEqual(expectedBlock, retBlock) {
-		t.Fatalf("Fail: got %+v\nexpected %+v", retBlock, expectedBlock)
+	require.Equal(t, expectedBlock, retBlock, "Could not validate returned retBlock as expected")
+
+}
+
+func TestAddBlock(t *testing.T) {
+	dataDir, err := ioutil.TempDir("", "TestAddBlock")
+	require.Nil(t, err)
+
+	//Create a new blockState
+	blockState, err := NewBlockState(dataDir)
+	require.Nil(t, err)
+
+	//Close DB
+	defer func() {
```

Comment:
Should we name this function and make it reusable? It would be nice to have a `blockState.Stop()` method that we use whenever starting the state service and then update this to `defer blockState.Stop()`. The error would then be logged in the `Stop` method and tested in its own `StartStateService` test. Thoughts?

Reply:
Sure but that will go a bit outside of the scope of this ticket.

Reply:
Adding a second test to the state service provides the opportunity to reuse some code in the tests. This was requested and carried over from the previous pull request. If you want to open a separate ticket for this that works too. This pattern is consistent with out other services and we should implement it at some point.

Reply:
We ofcourse should always think on re-using, I'll suggest we tackle this as part of the refactor on https://github.com/ChainSafe/gossamer/issues/530

Reply:
Can you update #530, make note in it, or open a new issue to address this?

Reply:
sure